rust/automerge-c/src/doc.rs

```diff
@@ -546,6 +546,31 @@ pub unsafe extern "C" fn AMobjSize(
     }
 }
 
+/// \memberof AMdoc
+/// \brief Gets the type of an object.
+///
+/// \param[in] doc A pointer to an `AMdoc` struct.
+/// \param[in] obj_id A pointer to an `AMobjId` struct or `AM_ROOT`.
+/// \return An `AMobjType`.
+/// \pre \p doc `!= NULL`.
+/// \internal
+///
+/// # Safety
+/// doc must be a valid pointer to an AMdoc
+/// obj_id must be a valid pointer to an AMobjId or std::ptr::null()
+#[no_mangle]
+pub unsafe extern "C" fn AMobjObjType(doc: *const AMdoc, obj_id: *const AMobjId) -> AMobjType {
+    if let Some(doc) = doc.as_ref() {
+        let obj_id = to_obj_id!(obj_id);
+        match doc.object_type(obj_id) {
+            None => AMobjType::Unknown,
```

Comment:
I think it might make more sense to return an optional `AMObjType` (I'm not sure how one does that idiomatically in modern C, just return a `NULL` pointer?). `automerge::Automerge::object_type` only returns `None` if whatever `obj_id` points at is not an object (either the `OpId` it wraps is not in the document, or it is a scalar value). It's not really unknown in any sense.

Reply:
I think a more typical style is what is done here (a sentinel value for unknown/invalid); pointers beget questions about ownership that don't apply in this case. maybe `AMobjType::Invalid` is the right phrasing?

Reply:
> I think it might make more sense to return an optional AMObjType (I'm not sure how one does that idiomatically in modern C, just return a NULL pointer?). 

@alexjg, yes, it would have to be NULL but then how would one free the storage for the `AMObjType` variant?
A variant doesn't complicate memory management like that.

Reply:
Yeah if it's idiomatic to return a sentinel value then maybe we just change the name? `Unknown` to me implies that there was an object there, we just didn't know what it was, when in fact we do know that there wasn't an object there.